gapic/src/main/com/google/gapid/views/PerformanceView.java

```diff
@@ -547,4 +598,167 @@ protected void okPressed() {
       }
     }
   }
+
+  private static class CounterDetailUi extends Composite {
+    private static final String ENTRY_LABEL_PREFIX = "Entry: ";
+    private static final String METRIC_LABEL_PREFIX = "Metric: ";
+
+    private final Label coordinateLabel;
+    private final CounterDetailPanel counterDetailPanel;
+    private final PanelCanvas canvas;
+    private final Map<Integer, Service.ProfilingData.Counter> counterDataLookup = Maps.newHashMap(); // metric id -> counter data.
+    private Service.ProfilingData.GpuCounters.Entry selectedEntry;      // Row.
+    private Service.ProfilingData.GpuCounters.Metric selectedMetric;    // Column.
+
+    public CounterDetailUi(Composite parent, Theme theme) {
+      super(parent, SWT.NONE);
+
+      setLayout(withMargin(new GridLayout(2, false), 0, 5));
+      Label titleLabel = withLayoutData(createLabel(this, "GPU Counter Detail Graph: "),
+          new GridData(SWT.LEFT, SWT.FILL, true, false, 2, 1));
+      titleLabel.setFont(theme.selectedTabTitleFont());
+      coordinateLabel = withLayoutData(createLabel(this, ENTRY_LABEL_PREFIX + "; " + METRIC_LABEL_PREFIX),
+          new GridData(SWT.LEFT, SWT.FILL, true, false));
+      withLayoutData(createLabel(this, "* (counter sample weight). ** Only for GPU counters."),
```

Comment:
Are these stars `*` supposed to appear in other labels in the UI (I didn't see them anywhere)? Or are they "footnotes" for the Detail Graph (in which case maybe we can just put this in parenthesis, without stars)?

Reply:
Yeah, my idea was to use it as a "footnotes", we may need a consistent UX for providing explanations I guess. And for the parenthesis, with `(counter sample weight)`, I was trying to say that in the graph, the number inside parenthesis is the counter weight, so there's both star and parenthesis. might not be a good footnote I guess...

Reply:
OK, thanks for the clarification. I think it's OK to merge like this, we can polish afterwards.